diff --git a/dist/es/motion/features/animation/exit.mjs b/dist/es/motion/features/animation/exit.mjs
index fe2506752d0dc652b71211a67e23061a49b87786..1c42384f3366fb975d58584266b4e748f869780f 100644
--- a/dist/es/motion/features/animation/exit.mjs
+++ b/dist/es/motion/features/animation/exit.mjs
@@ -11,7 +11,7 @@ class ExitAnimationFeature extends Feature {
             return;
         const { isPresent, onExitComplete } = this.node.presenceContext;
         const { isPresent: prevIsPresent } = this.node.prevPresenceContext || {};
-        if (!this.node.animationState || isPresent === prevIsPresent) {
+        if (!this.node.animationState || (isPresent === prevIsPresent && isPresent)) {
             return;
         }
         const exitAnimation = this.node.animationState.setActive("exit", !isPresent);
